spec/integration/federation/receive_federation_messages.rb

```diff
@@ -0,0 +1,224 @@
+require "spec_helper"
+require File.dirname(__FILE__) + "/factories"
+
+def generate_xml(entity, remote_user, user)
+  DiasporaFederation::Salmon::EncryptedSlap.generate_xml(
+    remote_user.diaspora_handle,
+    OpenSSL::PKey::RSA.new(remote_user.encryption_key),
+    entity,
+    OpenSSL::PKey::RSA.new(user.encryption_key)
+  )
+end
+
+shared_examples_for "it deals correctly with a relayable" do
+  before do
+    allow(DiasporaFederation.callbacks).to receive(:trigger)
+                                              .with(
+                                                :save_person_after_webfinger,
+                                                kind_of(DiasporaFederation::Entities::Person)
+                                              ).and_call_original
```

Comment:
When DiasporaFederation.callbacks is called with some arguments which weren't specified by expect/allow it fails, that's why. I put expect for a different callback, and because of this all other original callbacks fail to call.